core/src/main/java/org/elasticsearch/indices/memory/IndexingMemoryController.java

```diff
@@ -364,12 +360,15 @@ private int updateShardStatuses(EnumSet<ShardStatusChangeType> changes, List<Sha
             return changes;
         }
 
-        private void calcAndSetShardBuffers(int activeShards, String reason) {
-            if (activeShards == 0) {
+        private void calcAndSetShardBuffers(int activeShardCount, String reason) {
```

Comment:
> Hmm, confused: seems like we still need activeShardCount up front since we then divide total allowed heap by that?

what I mean that we should just loop on shardWasActive twice here, once to calculate activeShardCount and one to apply it. I think it will be clearer than passing it as an argument and trying to make sure it's correct in another functionn